### PR TITLE
Log endpoint connects/disconnects and put endpoint status message in JSON

### PR DESCRIPTION
**Pull request by Loonride**
A few useful additions for JSON log monitoring

Should we call it "endpoint_connected"/"endpoint_disconnected" or "endpoint_registered"/"endpoint_unregistered"?

### CODE REVIEW

**Comment by yadudoc** [comment]
There's a bit of nuance here. When the endpoint registers with the hosted services, the funcx-web-service calls the funcx-forwarder over REST to `register` the endpoint. We should log that. Next the endpoint having received the contact info for the forwarder sends a registration packet over ZMQ to the forwarder which is the point where the endpoint becomes `connected`. Whenever the funcx-forwarder fails to contact the endpoint over ZMQ, we mark the endpoint as `disconnected`.

**Comment by yadudoc** [comment]
We probably should add a state transition diagram here to better deduce the broken states for the forwarder and the endpoint.
Registered -> then nothing => connectivity issue over the 5500* ports
Registered -> Connected -> Registered => Forwarder down
Registered -> Connected -> Disconnected => Endpoint down  / Connection down
                                                                    Could be a transient network issue, and the endpoint could well be live with tasks/results.

File: funcx_forwarder/forwarder.py [hunk 1]
```diff
@@ -53,6 +53,20 @@ class Forwarder(Process):
...
+    - Registered: Endpoint has been registered with the service but has not yet connected
+    - Connected: Endpoint has connected, meaning ZMQ messages can be sent and received
+    - Disconnected: ZMQ message sending is failing for this endpoint, making it disconnected
+
+    Endpoint State Transitions
+    --------------------------
+    Registered -> Connected => Everything is working as expected (Endpoint is registered and ZMQ is working)
+    Registered -> nothing => Connectivity issue over the 5500* ports using ZMQ
+    Connected -> Registered => Forwarder down, endpoint re-registers
+    Connected -> Disconnected => Endpoint down / connection down (Could be transient network issue where endpoint is live with tasks/results)
```
Comment (Loonride) [inline funcx_forwarder/forwarder.py:69]: @yadudoc Is this what you had in mind? I slightly changed the diagram to indicate that Registered -> Connected is the normal transition, but transitions after that indicate something broke. Do you agree with that?

**Review by yadudoc — APPROVED** [review]
This looks good.

**Comment by yadudoc** [comment]
@Loonride, Let me know once you are done testing the changes.

**Comment by Loonride** [comment]
@yadudoc Tested this locally and it's good to go

**Pull request merged**